client/components/chart/utils.js

```diff
@@ -387,12 +405,18 @@ export const drawAxis = ( node, params ) => {
 			d3AxisBottom( xScale )
 				.tickValues( ticks )
 				.tickFormat( ( d, i ) => {
-					const monthDate = d instanceof Date ? d : new Date( d );
+					let monthDate = d instanceof Date ? d : new Date( d );
 					let prevMonth = i !== 0 ? ticks[ i - 1 ] : ticks[ i ];
 					prevMonth = prevMonth instanceof Date ? prevMonth : new Date( prevMonth );
-					return i === 0 || params.x2Format( monthDate ) !== params.x2Format( prevMonth )
-						? params.x2Format( monthDate )
-						: '';
+					monthDate =
+						i !== 0
+							? compareStrings(
+									params.x2Format( prevMonth ),
+									params.x2Format( monthDate ),
+									params
```

Comment:
I was testing this in master and it looks like something is not working fine.
![image](https://user-images.githubusercontent.com/3616980/47446657-fdb2a680-d7bb-11e8-91ab-4d488cc88a21.png)

I think the problem is we have to remove the third parameter here.

Reply:
Thanks @Aljullu I'll take a look.

Reply:
Fixed in https://github.com/woocommerce/wc-admin/pull/684